### PR TITLE
*: remove function "NewChunk"

### PR DESCRIPTION
**Pull request by zz-jason**
remove `NewChunk`, use `NewChunkWithCapacity` instead

### CODE REVIEW

**Comment by zz-jason** [comment]
@XuHuaiyu @winoros @lamxTyler @coocood PTAL

File: statistics/histogram.go [hunk 1]
```diff
@@ -577,7 +577,7 @@ func HistogramFromProto(protoHg *tipb.Histogram) *Histogram {
 
 func (hg *Histogram) popFirstBucket() {
 	hg.Buckets = hg.Buckets[1:]
-	c := chunk.NewChunk([]*types.FieldType{hg.tp, hg.tp})
+	c := chunk.NewChunkWithCapacity([]*types.FieldType{hg.tp, hg.tp}, hg.Bounds.NumRows()-2)
```
Comment (winoros) [inline statistics/histogram.go]: Why -2?

Comment (lamxTyler) [reply statistics/histogram.go]: Because the first bucket is poped, so the 2 boundaries are not needed.

**Comment by coocood** [comment]
LGTM

**Comment by jackysp** [comment]
Do we have some performance tests for this pr?

File: executor/distsql.go [hunk 2]
```diff
@@ -700,7 +702,7 @@ func (w *indexWorker) fetchHandles(ctx context.Context, result distsql.SelectRes
 			}
 		}
 	}()
-	chk := chunk.NewChunk([]*types.FieldType{types.NewFieldType(mysql.TypeLonglong)})
+	chk := chunk.NewChunkWithCapacity([]*types.FieldType{types.NewFieldType(mysql.TypeLonglong)}, w.maxChunkSize)
```
Comment (shenli) [inline executor/distsql.go]: Will this hurt the performance of OLTP queries?

**Comment by zz-jason** [comment]
@jackysp Performance has not done yet, I'll add one later.

**Pull request merged**